src/lib/protocols/IProtocol.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file IProtocol.ts
+ * @description Represents common properties and behavior of chains
+ */
+
+/**
+ * Transaction fee varies to control the transaction confirmation speed slow, normal and fast
+ */
+export interface SpendTransactionFee {
+  low: string;
+  average: string;
+  max: string;
+}
+
+export interface IProtocol {
```

Comment:
Have you considered to use abstract base class? Though I'm not sure about advantages comparing to `interface`

Reply:
I don't see any common behavior implementation among the protocols. Hence no abstract class. `interface` assuming we won't be relying on middlewares for these protocols in the future because otherwise these common middleware APIs could be part of these protocol files.

Reply:
@peronczyk renamed it to `IChainProtocol` and this file has been moved to the `types/` directory

Reply:
> I don't see any common behavior implementation among the protocols

This is why it is abstract, not a usual class.

Not sure how middleware is related to this topic.

As I know, interfaces a solution for multiple inheritance, but it is not the case here, so by default I would use an abstract class. You would get ability to use `instanceof ChainProtocol` in runtime and a simpler naming (without `I` 🙂).

Reply:
@davidyuk I will replace the interface with the Abstract class.

Reply:
The file has been moved here [src/protocols/BaseProtocol.ts](https://github.com/aeternity/superhero-wallet/pull/2193/files#diff-f6bde9421eb2156fa45d884fc21678748e1c4917a1138bf7d32a9182a0c9bdcd) 
@davidyuk Please have a look